h2o-py/tests/testdir_algos/gam/pyunit_PUBDEV_8877_gam_beta_constraints.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import division
+from __future__ import print_function
+from past.utils import old_div
+import sys
+sys.path.insert(1, "../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator
+
+# In this test, we check and make sure splines_non_negative should take precedence over the non_negative parameter
+def tea_gam_non_negative_spline_non_negative():
+    train1 = h2o.import_file(pyunit_utils.locate("smalldata/gam_test/nondecreasingCosTest.csv"))
+    gamX = ["a"]
+    x = ["a"]
+    numKnots = [4]
+    scale= [0.001]
+    bs_type = [2]
+    splines_non_negative = [False]
+    spline_order = [3]
+    # build GAM with non_negative set to False  The I-spline specification should take precedence
+    h2o_model = H2OGeneralizedAdditiveEstimator(gam_columns=gamX, scale=scale, bs=bs_type, spline_orders=spline_order,
+                                                num_knots=numKnots,  seed=12345, non_negative = True, 
```

Comment:
The non_negative should be set to false right?

Reply:
Non-negative is false by default.  I am trying to keep all the glm coefficients to be zero or positive while keeping all I-spline coefficients zero or negative.  This is what I am trying to make sure that works.

Reply:
Comment says `# build GAM with non_negative set to False` but `non_negative = True`

Reply:
Since I want to have positive or zero GLM coefficients, I need to set non_negative=True.  I changed the comment to the test as:

    # build GAM with non_negative set to True but splines_non_negative set to False.  The I-spline specification 
    # should take precedence